docs/_docs/rql.md

```diff
@@ -0,0 +1,658 @@
+  ["mtime", [">", "2020-01-01T22:15:52Z"]]]
+```
+
+This returns true for all entries whose `name` matches the glob `*.sh` OR the glob `*.json` AND whose `mtime` attribute is greater than `01/01/2020 10:15:52 PM UTC`. A query like this would be useful for finding files inside an AWS S3 bucket, a GCP Storage bucket, a container, or a VM. Specifically, you can use this query to find all `.sh` and `.json` files that were modified after `01/01/2020`.
+
+```
+["AND",
+  ["name", ["glob", "*.log]],
+  ["size", [">", 1024]]]
+```
+
+This returns true for all entries whose `name` matches the glob `*.log` AND whose `size` attribute is greater than 1024 bytes. You could use this to find all `.log` files that are greater than 1 KB (1024 bytes).
+
+```
+["AND",
+  ["kind", ["glob", "*container"]],
+  ["meta", ["object", [["key", "state"], "running"]]]]
+```
+
+This returns all entries whose `kind` matches the glob `*container` AND with `m['state'] == running`, where `m` is the entry's metadata. If the start path is `docker`, then this query would return all running Docker containers.
+
+See the [Primaries](#primaries) section for a list of all primaries and their documentation.
+
+## Entry schema optimization
+
+All RQL primaries are entry predicates. However some primaries can also be _entry schema_ predicates. Entry schema predicates act on an entry's schema; they are useful for optimizing RQL queries.
+
+Entry schema predicates are best illustrated by an example. Consider the following RQL query
+
+```
+["action", "exec"]
+```
+
+Assume the start path is `aws`, i.e. that we're starting at the AWS plugin. Then this query would return all entries in the AWS plugin that supported the `exec` action.
+
+Below is the AWS plugin's `stree` output.
+
+```
+wash . ❯ stree aws
+aws
+└── [profile]
+    └── resources
+        ├── s3
+        │   └── [bucket]
+        │       ├── [prefix]
+        │       │   ├── [prefix]
+        │       │   └── [object]
+        │       └── [object]
+        └── ec2
+            └── instances
+                └── [instance]
+                    ├── [console.out]
+                    ├── metadata.json
+                    └── fs
+                        ├── [dir]
+                        │   ├── [dir]
+                        │   └── [file]
+                        └── [file]
+```
+
+We will refer to this hierarchy when talking about the AWS plugin. EC2 instances are the only execable entries in this hierarchy. 
+
+The RQL will optimize its search to only recurse into entries that are execable or have an execable descendant. For the AWS plugin, this means that the RQL will _not_ recurse into S3 buckets (the node labeled `[bucket]`) or an EC2 instance's filesystem (the node labeled `fs`) -- those entries do not have any execable descendants. It will, however, recurse into AWS profiles (`[profile]`) and the resources directory (`resources`) since those entries have execable descendants (EC2 instances [`[instance]`]).
```

Comment:
Not sure what the double brackets in
```
EC2 instances [`[instance]`]
```
is supposed to imply. I think you could drop the outer brackets.

Reply:
Eh, just wanted to make it explicit. I agree though they aren't that necessary. Updated.